### PR TITLE
Limits: Extract header parsing into a util and add tests

### PR DESCRIPTION
**Pull request by joeruello**
Added some tests around the header parsing for checking limits.

### CODE REVIEW

File: ts-force/src/rest/rest.ts [hunk 1]
```diff
@@ -48,19 +47,15 @@ export class Rest {
...
-    private updateLimits (response: AxiosResponse) {
-        if (response.headers && response.headers['sforce-limit-info']) {
-            const [_, used, total] = LIMITS_REGEX.exec(response.headers['sforce-limit-info']);
-            this.apiLimit = {
-                used: Number(used),
-                limit: Number(total)
-            };
-        }
-        return response;
+        this.request.interceptors.response.use(updateLimits, updateLimits);
```
Comment (ChuckJonas) [inline ts-force/src/rest/rest.ts]: I don't think we should pass updateLimits into the second parameter of `use`.  It overrides the error handling and causes other tests to fail.

Think we just need: `this.request.interceptors.response.use(updateLimits`

Comment (joeruello) [reply ts-force/src/rest/rest.ts]: Updated

Comment (joeruello) [reply ts-force/src/rest/rest.ts]: Actually, can probally just inline the function and drop the assignment as well then.

**Comment by ChuckJonas** [comment]
Thanks for cleaning this up!  Going to go ahead and merge so I can test (and hopefully release soon)

**Comment by joeruello** [comment]
Cool, we'll be going ahead on our project next week, will be great to have this. Thanks for the support.

**Pull request merged**